src/libpgexporter/prometheus.c

```diff
@@ -1246,15 +1232,16 @@ custom_metrics(int client_fd)
 
    while (temp)
    {
-      if (temp->query_alt->is_histogram)
-      {
-         handle_histogram(store, &n_store, temp);
-      }
-      else
-      {
-         handle_gauge_counter(store, &n_store, temp);
+      if (temp->error || (temp->query != NULL && temp->query->tuples != NULL)) {
```

Comment:
`temp->error` ?

`{` on wrong line - uncrustify

Reply:
Yeah I'll use uncrustify and push it, forgot about this sorry!

Also, `temp` was sometimes a result of an error, which caused the queries in it to remain empty. Since it is a singly linked list, it was a bit of a problem to free it and reuse it again, and the cause of one of the issues I'd mentioned. Any query that resulted in an error (unsupported, etc.) gets cleaned up later.

I simply took a note of the return value of executing the query on the server (1 if error, 0 if not) and put it `temp->error`.